packages/verkle/src/verkleTree.ts

```diff
@@ -149,18 +159,28 @@ export class VerkleTree {
   /**
    * Gets a value given a `key`
    * @param key - the key to search for
-   * @param throwIfMissing - if true, throws if any nodes are missing. Used for verifying proofs. (default: false)
    * @returns A Promise that resolves to `Uint8Array` if a value was found or `null` if no value was found.
    */
-  async get(key: Uint8Array, throwIfMissing = false): Promise<Uint8Array | null> {
-    const node = await this.findLeafNode(key, throwIfMissing)
-    if (node !== null) {
-      const keyLastByte = key[key.length - 1]
-
-      // The retrieved leaf node contains an array of 256 possible values.
-      // The index of the value we want is at the key's last byte
-      return node.values?.[keyLastByte] ?? null
-    }
+  async get(key: Uint8Array): Promise<Uint8Array | null> {
+    verifyKeyLength(key)
+    const nodeRLP = await this._db.get(key)
+    if (nodeRLP === undefined) return null
+    // We first retrieve the node and decode it
+    const rawNode = RLP.decode(nodeRLP) as Uint8Array[]
+    const node =
+      rawNode[0][0] === VerkleNodeType.Leaf
+        ? // TODO: Figure out how to determine depth from key
```

Comment:
I don't think Depth can be directly inferred from the key, as the same key would have different depths depending on which verkle trie structure they are a part of. Shouldn't that information be contained in the serialized node that is stored in the db? 

I just checked the go code and they don't store that in their (legacy) serialization format..  maybe I'm missing something.

Reply:
Okay, I think that makes sense to me.  We can ensure that's included in the serialized node.